src/components/common/ColonyActions/ActionDetailsPage/DefaultMotion/MotionPhaseWidget/VotingWidget/useVotingWidgetUpdate.ts

```diff
@@ -19,6 +19,14 @@ export const useVotingWidgetUpdate = (
     setHasUserVoted(true);
   }
 
+  useEffect(() => {
```

Comment:
Ideally we'd express this as:

``` js
useEffect(() => {
setHasUserVoted(!!currentVotingRecord);
}, [currentVotingRecord]
```

However, it doesn't satisfy the end case where a user votes for the first time, and then signs out before the db has been updated. In that case, the `currentVotingRecord` will start as undefined, and remain undefined, which is why I'm capturing the `!user` logic explicitly.